### ISSUE TITLE
Rename Hessenberg to HessenbergFactorization

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaLang/julia#23502]
The following is thoroughly confusing:
```julia
julia> a = rand(3,3)
3×3 Array{Float64,2}:
 0.490033  0.522445  0.854002
 0.309698  0.557504  0.495107
 0.980559  0.81369   0.961208

julia> LinAlg.Hessenberg(a)|>full
3×3 Array{Float64,2}:
 0.490033  0.522445  0.854002
 0.309698  0.557504  0.495107
 0.980559  0.81369   0.961208

julia> hessfact(a)[:H] # returns a Matrix{Float64}
3×3 Array{Float64,2}:
 0.490033   0.651441  -0.760193
 1.26273    0.539174  -0.807753
 0.0       -0.48917    0.979538
```

One would expect `Hessenberg(a)` to work analogously to `Symmetric(a)`.  

See also discussion https://github.com/JuliaMath/Polynomials.jl/issues/130#issuecomment-325762188

**Comment by ViralBShah** [comment]
I think we are done with name changes for 1.0 and this is consistent with the others. Perhaps it is ok to close?

**Comment by andreasnoack** [comment]
> this is consistent with the others

The problem here is that the factorization and a matrix involved in the factorization share the same name. I don't think that is the case for any other matrix factorization. I think we should just keep this one open even though it probably won't be fixed in 1.0.
